incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/MqttKafkaConfiguration.java

```diff
@@ -105,106 +110,105 @@ public int willStreamReconnectDelay()
         Configuration config,
         String fullyQualifiedMethodName)
     {
-        Supplier<String> supplier = null;
+        StringSupplier supplier = null;
 
         try
         {
-            MethodType signature = MethodType.methodType(String.class);
+            MethodType signature = MethodType.methodType(StringSupplier.class);
             String[] parts = fullyQualifiedMethodName.split("::");
             Class<?> ownerClass = Class.forName(parts[0]);
             String methodName = parts[1];
             MethodHandle method = MethodHandles.publicLookup().findStatic(ownerClass, methodName, signature);
             supplier = () ->
             {
-                String value = null;
+                StringSupplier value = null;
                 try
                 {
-                    value = (String) method.invoke();
+                    value = (StringSupplier) method.invoke();
                 }
                 catch (Throwable ex)
                 {
                     LangUtil.rethrowUnchecked(ex);
                 }
 
-                return value;
+                return value.get();
             };
         }
         catch (Throwable ex)
         {
             LangUtil.rethrowUnchecked(ex);
         }
 
-        return supplier.get();
+        return supplier;
     }
 
-    private static long decodeLongSupplier(
+    private static LongSupplier decodeLongSupplier(
         Configuration config,
         String fullyQualifiedMethodName)
     {
         LongSupplier supplier = null;
 
         try
         {
-            MethodType signature = MethodType.methodType(long.class);
+            MethodType signature = MethodType.methodType(LongSupplier.class);
             String[] parts = fullyQualifiedMethodName.split("::");
             Class<?> ownerClass = Class.forName(parts[0]);
             String methodName = parts[1];
             MethodHandle method = MethodHandles.publicLookup().findStatic(ownerClass, methodName, signature);
             supplier = () ->
             {
-                long value = 0;
+                LongSupplier value = null;
                 try
                 {
-                    value = (long) method.invoke();
+                    value = (LongSupplier) method.invoke();
                 }
                 catch (Throwable ex)
                 {
                     LangUtil.rethrowUnchecked(ex);
                 }
 
-                return value;
+                return value.getAsLong();
             };
         }
         catch (Throwable ex)
         {
             LangUtil.rethrowUnchecked(ex);
         }
 
-        return supplier.getAsLong();
+        return supplier;
     }
 
-    private static String defaultInstanceId(
+    private static StringSupplier defaultInstanceId(
         Configuration config)
     {
-        return String.format("%s-%s", "zilla", UUID.randomUUID());
+        return () -> String.format("%s-%s", "zilla", UUID.randomUUID());
```

Comment:
Previous was correct. The default method is a supplier, does not return a supplier.
Same for other `StringSupplier` and `LongSupplier` defaults.